File: Aufgabe1/solutions/1.2/sqrt.py


def sqrt_newton(x):
  EPS = 0.000001
  MI  = 10000

  iter     = 0
  estimate = 1.0

  while (abs(estimate*estimate - x) > EPS and iter < MI):
    estimate = 0.5*(estimate + x/estimate)
    iter = iter + 1

  return estimate

x = input("Bitte geben Sie eine Zahl ein: ")
print("Die Wurzel aus " + x + " ist: " + str(sqrt_newton(float(x))))

File: Aufgabe1/solutions/1.2/sqrt.js

function sqrt_newton(x) {
  const EPSILON  = 0.000001;
  const MAX_ITER = 10000;

  let iter     = 0;
  let estimate = 1.0;

  while (Math.abs(estimate*estimate - x) > EPSILON && iter < MAX_ITER) {
    estimate = 0.5*(estimate + x/estimate);
    ++iter;
  }

  return estimate;
}

// Node.js stuff
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Bitte geben Sie eine Zahl ein:', (x) => {
  console.log(`Die Wurzel aus ${x} ist ${sqrt_newton(x)}`);
  rl.close();
});